spec/helpers/react_helper_spec.rb

```diff
@@ -38,6 +39,54 @@
       end
     end
 
+    context 'when alert is present and has text and title' do
+      let(:alert) { instance_double(Alert, text: 'alert_title', title: 'alert_title') }
+
+      it 'includes alert and has text and title' do
+        helper.search_results_layout(search, {}, vertical, affiliate)
+        expect(helper).to have_received(:react_component).with(
+          'SearchResultsLayout',
+          hash_excluding(:alert)
+        )
+      end
+    end
+
+    context 'when alert is present but text is blank' do
+      let(:alert) { instance_double(Alert, text: '', title: 'alert_title') }
```

Comment:
Rather than an instance double it's better to use the actual object. In this case I think something like will do the trick.

```ruby
before do
   affiliate.alert.text = ''
end
```